### PR TITLE
Combo

### PR DESCRIPTION
**Pull request by noahadams**
Integrates use of combo with localStorage based caching, and some tests of said feature

### CODE REVIEW

**Comment by johnboxall** [comment]
`$.fn.combineScripts` should probably handle the cases where: No scripts were found / No external scripts were found.

Also, let's say `localStorage` isn't available - are we looking at a performance penalty? If so, if persistence isn't available, maybe we should just return the scripts we find in order.

File: api/combo-caching.js [hunk 1]
```diff
@@ -164,12 +175,15 @@ var ccPublic = /^\s*public\s*$/
                 }
             }
 
+            //DEBUG
+            // console.log('urls: ' + JSON.stringify(urls));
+            // console.log('notCachedUrls: ' + JSON.stringify(notCachedUrls));
+
```
Comment (rrjamie) [inline api/combo-caching.js]: Remove // Debug comments if possible.

**Comment by rrjamie** [comment]
What's the status of this?

**Comment by johnboxall** [comment]
@rrjamie the latest code is awaiting review to be merged.

File: api/combo.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+                    (!cacheControl['no-cache'])) {
+                    // Convert the max-age directive to ms.
+                    return now > (date + (cacheControl['max-age'] * 1000));
+                }
+            }
+
+            // If `expires` is present, we are stale if we are older.
+            if (expires) {
+                return now > Date.parse(expires);
+            }
```
Comment (rrjamie) [inline api/combo.js:206]: Is this what we want? For non-parsables to never be stale:

```
 >>> 1 > Date.parse("Barf Barf Barf")
 false
```

File: api/combo.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+            // Otherwise, we are stale.
+            return false;
```
Comment (rrjamie) [inline api/combo.js]: If we are stale, return true?

**Pull request merged**